roles/swift-common/handlers/main.yml

```diff
@@ -1,12 +1,24 @@
 ---
 - name: pip install swifttool
-  pip: name="/opt/stack/{{ swift.swifttool_dir  }}" extra_args='-i https://pypi.python.org/simple/'
+  pip: name="/opt/stack/{{ swift.swifttool_dir  }}" extra_args="-i https://pypi.python.org/simple/"
+       virtualenv={{ swift.venv_dir }}
   register: result
   until: result|success
   retries: 3
   delay: 10
   notify:
     - update ca-certs
+    - pip clean swifttool outside venv
+    - setup swifttool alternatives
+
+- name: pip clean swifttool outside venv
+  pip: name=swifttool state=absent
+  ignore_errors: True
```

Comment:
Why would we need to ignore errors here? If the package doesn't exist, the task will return unchanged. If it does, and we have some failure removing it, we need to know about it, otherwise the system will be in an unknown state.